PiwikTracker/Queue.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  EventQueue.swift
+//  PiwikTracker
+//
+//  Created by Cornelius Horstmann on 09.10.16.
+//  Copyright © 2016 Mattias Levin. All rights reserved.
+//
+
+import Foundation
+
+/// An Element that is queueable.
+protocol Queueable {
+    /// A unique identifier. It should be unique per object.
+    var id: UUID { get }
+    
+    /// This property is used to get the data that should be stored in the queue.
+    var data: Data { get }
+    
+    /// This initializer is used when dequeueing.
+    init?(data: Data)
+}
+
+/// A queue in which Objects that conform to queueable can be stored.
+protocol Queue {
+    /// The Item type have to conform to the Queueable protocol.
+    associatedtype Item: Queueable
```

Comment:
I would maybe try/start without the `Queueable` protocol and only add it later if needed.

Maybe Item can conform to `NSCoding`?
You should be able to do exactly the same as with the `Queueable` (but I think it will force the object to be a class)